webrender/src/render_backend.rs

```diff
@@ -204,10 +207,15 @@ impl Document {
                     }
                 };
 
+                let scrolled = self.scroll_nearest_scrolling_ancestor(delta, node_index);
+
                 let should_render =
                     should_render &&
-                    self.scroll_nearest_scrolling_ancestor(delta, node_index) &&
+                    scrolled &&
                     self.render_on_scroll == Some(true);
+
+                self.render_on_generate_frame |= scrolled && !should_render;
```

Comment:
could you explain the logic here please?

Reply:
The comment just above this about double borrowing is probably contributing to this being more complex than it should be.

That said, I think we could rename the earlier usage of should_render to found_scroll_node, and then make this code look like:

let mut should_render = false;
if (found_scroll_node && self.scroll_nearest_scrolling_ancestor(delta, node_index) {
  if (self.render_on_scroll == Some(true)) {
    should_render = true;
  } else {
    self.render_on_generate_frame = true;
  }
}

Reply:
(heads up - I edited your code a bit)
the suggestion sounds good to me

Reply:
Thanks @mattwoodrow, it is what I intended!